lib/Db/TagMapper.php

```diff
@@ -248,4 +225,31 @@ public function createDefaultTags(MailAccount $account): void {
 			$this->insert($entity);
 		}
 	}
+
+	public function deleteDuplicates(): void {
+		$selectQb = $this->db->getQueryBuilder();
```

Comment:
Unfortunately this won't work in MySQL and probably others, because you can't update a table that is used in a select. I'll have to loop through the results of the subquery.

Reply:
OK. Executing the subselect, fetching the results and use it for the delete operation should also work. There is a check if more than 1000 arguments passed to a query. Need to limit the number of mail_message_tags in one delete operation to 1000.